### PR TITLE
Understand variadic arg types with `@no-named-arguments`

### PR DESCRIPTION
**Pull request by jrmajor**
First step for phpstan/phpstan#5968.

### CODE REVIEW

File: src/Analyser/MutatingScope.php [hunk 1]
```diff
@@ -2610,7 +2614,7 @@ private function enterFunctionLike(
 		foreach (ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getParameters() as $parameter) {
 			$parameterType = $parameter->getType();
 			if ($parameter->isVariadic()) {
-				if ($this->phpVersion->supportsNamedArguments()) {
+				if ($this->phpVersion->supportsNamedArguments() && $functionReflection->acceptsNamedArguments()) {
```
Comment (jrmajor) [inline src/Analyser/MutatingScope.php:2617]: There is a simillar check in [`getFunctionType()`](https://github.com/phpstan/phpstan-src/blob/cf8832c79b51cc3d8751b1f3651cfcf31ec5298c/src/Analyser/MutatingScope.php#L3051-L3065). I'm not sure whether I need to change anything there, as it seems to work as is.

**Comment by jrmajor** [comment]
We would also need a rule preventing child class from declaring `@no-named-arguments` when the parent doesn't and a rule reporting usage of named arguments when this annotation is present.

**Comment by ondrejmirtes** [comment]
Really really nice!

> We would also need a rule preventing child class from declaring...

This should be done in MethodSignatureRule.

And the usage of named arguments in `@no-named-arguments` should be reported in FunctionCallParametersCheck.

**Comment by ondrejmirtes** [comment]
Thank you! Looking forward to follow-up PRs!

**Comment by mhsdesign** [comment]
Fyi this seems to be the first followup https://github.com/phpstan/phpstan-src/pull/1400 but there seems to be no pr yet for preventing the calling site to use named args

**Pull request merged**